Mathlib/Data/Finset/Image/Defs.lean

```diff
@@ -0,0 +1,390 @@
+    (h : Function.Semiconj f ga gb) : Function.Semiconj (map f) (map ga) (map gb) := fun _ =>
+  map_comm h
+
+theorem _root_.Function.Commute.finset_map {f g : α ↪ α} (h : Function.Commute f g) :
+    Function.Commute (map f) (map g) :=
+  Function.Semiconj.finset_map h
+
+@[simp]
+theorem map_subset_map {s₁ s₂ : Finset α} : s₁.map f ⊆ s₂.map f ↔ s₁ ⊆ s₂ :=
+  ⟨fun h _ xs => (mem_map' _).1 <| h <| (mem_map' f).2 xs,
+   fun h => by simp [subset_def, Multiset.map_subset_map h]⟩
+
+@[gcongr] alias ⟨_, _root_.GCongr.finsetMap_subset⟩ := map_subset_map
+
+/-- The `Finset` version of `Equiv.subset_symm_image`. -/
+theorem subset_map_symm {t : Finset β} {f : α ≃ β} : s ⊆ t.map f.symm ↔ s.map f ⊆ t := by
+  constructor <;> intro h x hx
+  · simp only [mem_map_equiv, Equiv.symm_symm] at hx
+    simpa using h hx
+  · simp only [mem_map_equiv]
+    exact h (by simp [hx])
+
+/-- The `Finset` version of `Equiv.symm_image_subset`. -/
+theorem map_symm_subset {t : Finset β} {f : α ≃ β} : t.map f.symm ⊆ s ↔ t ⊆ s.map f := by
+  simp only [← subset_map_symm, Equiv.symm_symm]
+
+/-- Associate to an embedding `f` from `α` to `β` the order embedding that maps a finset to its
+image under `f`. -/
+def mapEmbedding (f : α ↪ β) : Finset α ↪o Finset β :=
+  OrderEmbedding.ofMapLEIff (map f) fun _ _ => map_subset_map
+
+@[simp]
+theorem map_inj {s₁ s₂ : Finset α} : s₁.map f = s₂.map f ↔ s₁ = s₂ :=
+  (mapEmbedding f).injective.eq_iff
+
+theorem map_injective (f : α ↪ β) : Injective (map f) :=
+  (mapEmbedding f).injective
+
+@[simp]
+theorem map_ssubset_map {s t : Finset α} : s.map f ⊂ t.map f ↔ s ⊂ t := (mapEmbedding f).lt_iff_lt
+
+@[gcongr] alias ⟨_, _root_.GCongr.finsetMap_ssubset⟩ := map_ssubset_map
+
+@[simp]
+theorem mapEmbedding_apply : mapEmbedding f s = map f s :=
+  rfl
+
+end Map
+
+/-! ### image -/
+
+
+section Image
+
+variable [DecidableEq β]
+
+/-- `image f s` is the forward image of `s` under `f`. -/
+def image (f : α → β) (s : Finset α) : Finset β :=
+  (s.1.map f).toFinset
+
+@[simp]
+theorem image_val (f : α → β) (s : Finset α) : (image f s).1 = (s.1.map f).dedup :=
+  rfl
+
+variable {f g : α → β} {s : Finset α} {t : Finset β} {a : α} {b c : β}
+
+@[simp]
+theorem mem_image : b ∈ s.image f ↔ ∃ a ∈ s, f a = b := by
+  simp only [mem_def, image_val, mem_dedup, Multiset.mem_map, exists_prop]
+
+theorem mem_image_of_mem (f : α → β) {a} (h : a ∈ s) : f a ∈ s.image f :=
+  mem_image.2 ⟨_, h, rfl⟩
+
+lemma forall_mem_image {p : β → Prop} : (∀ y ∈ s.image f, p y) ↔ ∀ ⦃x⦄, x ∈ s → p (f x) := by simp
+lemma exists_mem_image {p : β → Prop} : (∃ y ∈ s.image f, p y) ↔ ∃ x ∈ s, p (f x) := by simp
+
+@[deprecated (since := "2024-11-23")] alias forall_image := forall_mem_image
+
+theorem map_eq_image (f : α ↪ β) (s : Finset α) : s.map f = s.image f :=
+  eq_of_veq (s.map f).2.dedup.symm
+
+@[norm_cast]
+theorem image_eq_map (f : α ↪ β) (s : Finset α) : s.image f = s.map f :=
+  (map_eq_image f s).symm
```

Comment:
Is this new? Why?

Reply:
Yes, without this the `mod_cast` in 
```
theorem map_union [DecidableEq α] [DecidableEq β] {f : α ↪ β} (s₁ s₂ : Finset α) :
    (s₁ ∪ s₂).map f = s₁.map f ∪ s₂.map f :=
  mod_cast Set.image_union f s₁ s₂

theorem map_inter [DecidableEq α] [DecidableEq β] {f : α ↪ β} (s₁ s₂ : Finset α) :
    (s₁ ∩ s₂).map f = s₁.map f ∩ s₂.map f :=
  mod_cast Set.image_inter f.injective (s := s₁) (t := s₂)
```
in `Mathlib.Data.Finset.Image.Basic` both fail.